File: go/greeter/greeter_test.go
package greeter_test

import (
	"testing"

	"github.com/mike-boquard/bazel-intro/go/greeter"
)

func TestHello(t *testing.T) {
	cases := []struct {
		input string
		want  string
	}{
		{"World", "Hello, World!"},
		{"", "Hello, !"},
		{"日本語", "Hello, 日本語!"},
	}
	for _, c := range cases {
		got := greeter.Hello(c.input)
		if got != c.want {
			t.Errorf("Hello(%q) = %q, want %q", c.input, got, c.want)
		}
	}
}

File: cpp/greeter/greeter_test.cc
#include "cpp/greeter/greeter.h"
#include "gtest/gtest.h"

TEST(GreeterTest, HelloWithName) { EXPECT_EQ(greeter::Hello("World"), "Hello, World!"); }

TEST(GreeterTest, HelloWithEmptyString) { EXPECT_EQ(greeter::Hello(""), "Hello, !"); }

TEST(GreeterTest, HelloWithUnicode) { EXPECT_EQ(greeter::Hello("日本語"), "Hello, 日本語!"); }
